src/swagger_endpoints.erl

```diff
@@ -216,13 +216,17 @@ jesse_json_schema(Schema, _Options) ->
   #{error => Schema}.
 
 get_version(Doc) ->
-    case proplists:get_value("swagger", Doc, not_set) of
+    case oas_base_version(proplists:get_value("swagger", Doc, not_set)) of
         ?OAS2 -> ?OAS2;
         not_set ->
-            case proplists:get_value("openapi", Doc) of
+            case oas_base_version(proplists:get_value("openapi", Doc)) of
                 ?OAS3 -> ?OAS3
             end
     end.
 
 definitions_prefix(?OAS2) -> "/definitions/";
 definitions_prefix(?OAS3) -> "/components/schemas/".
+
+oas_base_version("2.0")       -> ?OAS2;
+oas_base_version("3.0." ++ _) -> ?OAS3;
+oas_base_version(V)           -> V.
```

Comment:
Shouldn't we crash (log error) on the 3rd case?

Reply:
I double this one: if we don't crash here, we would crash ungracefully on the very first pattern match that expects one of the two macros.